kvui.py

```diff
@@ -299,6 +300,39 @@ def apply_selection(self, rv, index, is_selected):
         self.selected = is_selected
 
 
+class AutocompleteHintInput(TextInput):
+    min_chars = NumericProperty(3)
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self.dropdown = DropDown()
+        self.dropdown.bind(on_select=lambda instance, x: setattr(self, 'text', x))
+        self.bind(on_text_validate=self.on_message)
+
+    def on_message(self, instance):
+        App.get_running_app().commandprocessor("!hint "+instance.text)
+
+    def on_text(self, instance, value):
+        if len(value) >= self.min_chars:
+            self.dropdown.clear_widgets()
+            ctx: context_type = App.get_running_app().ctx
+            item_names = ctx.item_names._game_store[ctx.game].values()
+
+            def on_press(button: Button):
+                return self.dropdown.select(button.text)
+            lowered = value.lower()
+            for item_name in item_names:
+                if lowered in item_name.lower():
```

Comment:
Recommend using `.casefold()` instead of `.lower()` here. I doubt we have any item names where this makes a difference, but it's probably better to play it safe seeing as there's nothing stopping such item names from being added in the future.

Reply:
The only difference I can see is that casefold is a more complex, slower algorithm.

Reply:
Nope, they're same speed. The thing is both casefold() and lower() come from the unicode standard, and thus they must operate on all unicode edgecases. I think casefold may actually be slightly faster:

```python
>>> def my_casefold():
...   return 'The quick BROWN fox jumps over the LAZY dog'.casefold()
...
>>> def my_lower():
...   return 'The quick BROWN fox jumps over the LAZY dog'.lower()
...
>>> timeit.timeit(my_lower, number=100000)
0.011489199940115213
>>> timeit.timeit(my_lower, number=10000000)
1.1040864000096917
>>> timeit.timeit(my_casefold, number=10000000)
1.078911299817264
```

Reply:
what's the main difference? German sharp S (`ß') casefolding to `ss`?

Reply:
For me the important thing is communicating purpose. [this link](https://www.w3.org/TR/charmod-norm/#definitionCaseFolding) seems to explain more, gives the sharp s example and a Greek example as well, I assume there's more. Also covers the Turkish I problem, which is not covered by Python's casefold (I don't think we care that much about Turkish support though).

As far as I'm concerned, this is a free change that will make things behave better if any item names use different scripts, and zero additional performance cost (or even performance improvement if my measurements above also apply to those examples).